### ISSUE TITLE
Proposal:PDF Viewer with support for search, annotations and digital signature validation.

### ISSUE THREAD

**Issue by applefanbois** [microsoft/microsoft-ui-xaml#2496]
It is no secret that developing a business software for Windows is very costly compared to Android, iOS and macOS because they have almost zero support for PDF in the native SDK. So you have to pay 10 000$us per year for 3rd party SDK just to display a simple pdf in your native app while on other platforms they have fully functionals tools in the SDK for free.

Our app like all modern accounting app require a lot of pdf manipulations. So It would be nice to have the same tools has other more modern platforms : macOS, iOS, Android.

on iOS the Webview can even open xls, docx... on UWP, you can't do that.
Another 10 000$us per year to display xls, docx?

No wonder there is no new app made fully natively for windows since 2006.

Also a way to decompose, reassemble, reorder pages add new pdf pages from images would be a nice to have. The other platforms have these functionality since ... a long time ago. 

All of this should work in UWP, because our app is new, secure and respectful of privacy and the old apps are not (they have full access).

By the time MS finish wasting time with deprecated win32, Apple and Android will be too far ahead to catch.

**Comment by applefanbois** [comment]
For reference, the Apple PDFKit should be the base reference
https://developer.apple.com/documentation/pdfkit

At minimum, WinUI should have all the exact same features. But it would not be difficult to do improve and make it simpler to use.

**Comment by applefanbois** [comment]
Android SDK has a much less powerful pdf tools in their SDK than Apple, but it is still miles ahead of windows native SDK.
https://developer.android.com/reference/kotlin/android/graphics/pdf/package-summary

Just look at this miserable pdf support on Windows
https://docs.microsoft.com/en-us/uwp/api/windows.data.pdf?view=winrt-18362

PDF is as important as webview for business app.

**Comment by applefanbois** [comment]
We need to be able to create, edit and view (with text search and selection) pdf.

**Comment by Patfreeze** [comment]
Any progress ?

**Comment by applefanbois** [comment]
No, we tried many times and failed. This WebView2 won't run on UWP. We waited 10 years for a decent webview, we can wait another year. Our iOS/macOS/Android app has been released a while ago, but the Windows version will have to wait another year. We tell our client to invest in Apple product because it will always be the most advanced version because MS is not giving the dev decent SDK compared to Apple and Google.

**Comment by Herrick19** [comment]
Our company is refreshing our accounting software from win32 to UWP. Our old software use a very old and limited 3rd party library made in C++ (This library has no support anymore). So for the pas 4 years, we have made a newer version of our app for mobile platforms because there was a huge demand for it. Now it is time to port back this new app to all Windows 10 devices : hololens, Windows ARM, good old PC. We want to be compatible with everything, after all this is our future, we want to make one app that will run on devices that our customers use. Half of our customers works on the road, the other half in an office.

Sadly, we have quickly found ourselves at a dead end on Windows UWP. The webview is so limited, that only 1/5 of the mobile functionality version will be supported on Windows 10. Webview2 is not working right now and won't be until 1 year maybe? So our Windows 10 app will be a lot less powerful than the mobile version. In fact, it is almost useless. It even manage to have less features that our old win32 app.

Back on the subject, the best we could do right now is to use the only native UWP SDK pdf tools to render a pdf page to an image and display that image. Not only the viewer has no search, no text selection for copy/paste; but it is impossible to edit and create pdf. Web and PDF are the central part of our app. All the logic is in the cloud. We are an accounting application, our customers work with digitally signed PDF.
Can we say to our customers, you can do that on your phone or mac, but on Windows, you have to buy Adobe Pro?

On Android and iOS, the webview can display xls, docx and pdf, on UWP it is not possible, so there is no other way for now. We could pay for a 3rd party PDF SDK, but the ones that are truly UWP cost thousands of dollars per year.

We look at our competitors, they are all 100% web based. So they are totally independent of Windows and Microsoft products. We try to give an edge to our customers with more functionality that a regular browser cannot offer. Sadly, we have to explain to our customers that our brand new shiny and feature rich app is only available on macOS, iOS and Android. Windows version won't be released until another year maybe. We have to tell them that for now, all we have for them is the old limited win32 version unless they by a mac.

The lack of proper webview and pdf support for native app has painted Windows 10 in a corner. When looking on the internet for solution, we realized that we are the only company making a new app for Windows. The others are doing things based on Web or QT. There is no native dev left on Windows, even the game industry is using Unreal and Unity (they are platform independent). Why is that? Because MS has abandoned the idea of native app. Weird because Android, iOS and soon Huawei are thriving on native app.

Please MS, add proper PDF and Web support to your SDK. Without these 2 tools, Windows Dev is a derelict platform. And please fix the store, we don't want to manage installer and updates ourselves. Once you experience the mobile DEV and infrastructure, you don't want to go back to the past that MS is offering.

**Comment by JaiganeshKumaran** [comment]
@applefanbois I'm late to respond but Windows does actually have a very barebones PDF API under the [Windows.Data.Pdf namespace](https://learn.microsoft.com/en-us/uwp/api/windows.data.pdf?view=winrt-22621). All it can do is render PDF pages as a set of images, which should do the job for a simple viewer but can't really support the features you're requesting.
